tests/discovery/test_openreview.py

```diff
@@ -0,0 +1,110 @@
+    query_params = {**query_params, "block_size": 100, "limit": 1000}
+    openreview_dispatch = config.discoverers["openreview"]
+    api_versions: list[int] = openreview_dispatch.api_versions
+
+    papers_per_version: dict[int, list[PaperInfo]] = {}
+    for api_version in api_versions:
+        discoverer = OpenReview(api_version)
+
+        try:
+            papers_per_version[api_version] = sorted(
+                discoverer.query(**query_params),
+                key=lambda x: x.paper.title,
+            )
+        except openreview.openreview.OpenReviewException:
+            papers_per_version[api_version] = []
+
+    if papers_per_version[1] and papers_per_version[2]:
+        assert False, (
+            f"The same papers are not expected to be in version 1 and 2 at the same time. Papers: {len(papers_per_version[1])=} {len(papers_per_version[2])=}"
+        )
+
+    papers: list[PaperInfo] = sum(papers_per_version.values(), [])
+
+    assert papers, f"No papers found for {query_params=}"
+
+    assert [p.paper for p in papers] == [
+        p.paper
+        for p in sorted(
+            openreview_dispatch.query(**query_params),
+            key=lambda x: x.paper.title,
+        )
+    ], (
+        f"Querying with OpenReview({api_versions=}) should return the same papers as querying with OpenReviewDispatch"
+    )
+
+    match next(iter(query_params.keys())):
+        case "venue":
+            assert all(
+                any(
+                    query_params["venue"].lower() == rel.venue.name.lower()
+                    for rel in iter_releases(paper.paper)
+                )
+                for paper in papers
+            ), f"Some papers do not contain the venue {query_params['venue']=}"
+        case "paper_id":
+            assert all(
+                any(
+                    query_params["paper_id"] == link_id
+                    for link_id in iter_links_ids(paper.paper)
+                )
+                for paper in papers
+            ), f"Some papers do not contain the paper ID {query_params['paper_id']=}"
+        case "author":
+            assert all(
+                any(
+                    query_params["author"].lower() in author.author.name.lower()
+                    for author in paper.paper.authors
+                )
+                for paper in papers
+            ), f"Some papers do not contain the author {query_params['author']=}"
+        case "author_id":
+            assert [p.paper for p in papers] == [
+                p.paper
+                for p in sorted(
+                    openreview_dispatch.query(
+                        author="Yoshua Bengio",
+                        venue=query_params["venue"],
+                        block_size=100,
+                        limit=100,
+                    ),
+                    key=lambda x: x.paper.title,
+                )
+            ], (
+                "Querying by author ID should return the same papers as querying by author name"
```

Comment:
I'm not sure that's true, strictly speaking. We can probably assume it's true for Yoshua, but in some cases multiple IDs might map to the same author name.

Reply:
Ah so should we delete this test? I've updated the message but maybe we can simply delete the test

Reply:
We do want to make sure search by author_id works. I think we can keep it like this for now, if it breaks later we can fix something up easily enough.